src/ocl/convolutionocl.cpp

```diff
@@ -826,55 +827,98 @@ void ConvolutionDescriptor::GetSolutionsFallback(Handle& handle,
     ctx.SetStream(&handle);
     ctx.DetectRocm();
 
-    const auto wti2time = [](const float& wti) {
-        assert(wti != 0.0f);
-        if(wti <= 0.0f) // Return negative values as is, avoid DIV/0.
-            return wti;
-        return 10.0f / wti; // Assume WTI == 1.0 (100%) is 10 ms.
-    };
-
-    for(const auto& solver_id : solver::GetSolversByPrimitive(solver::Primitive::Convolution))
+    if(MIOPEN_ENABLE_HEUR && ConvHeur::IsApplicable(handle.GetDeviceName(), problem.conv_problem))
     {
-        // solver_id is always valid here, because taken from registry.
-        // Validity check is not required.
-        const auto algo = solver_id.GetAlgo();
-        if(IsAlgorithmDisabled(algo)) // Algos can be disabled globally.
-            continue;
-        const auto& s = solver_id.GetSolver();
-        if(s.IsEmpty())
-            continue;
-        if(!s.IsDynamic()) // Let's allow non-dynamic later, if necessary.
-            continue;
-        if(!s.IsApplicable(ctx))
-            continue;
+        int idx = 1; // Each solution to have a successively more negative values keeping
+        // sorting logic intact in frameworks
+        bool is_cached = false;
+        auto solvers = ConvHeur{}.Estimate(handle.GetDeviceName(), problem.conv_problem, is_cached);
```

Comment:
@Dmantri98 Please share the time cost numbers collected for this function to answer @atamazov's query.

Reply:
Inital run is around 8ms (needs to load model into Frugally-Deep, parse metadata file, and perform inference). Subsequent runs are around 2ms.

Reply:
@atamazov  Please note this data point, if you need additional metrics we can share

Reply:
@Dmantri98 Please ping me when AI heuristics is ready for gfx906 or gfx1030 and I will do performance testing.

Reply:
I don't believe we will be implementing the AI Heuristic for gfx906, and the data collection and model development for gfx1030 has not been done yet.

Reply:
@JehandadKhan @junliume 
> I don't believe we will be implementing the AI Heuristic for gfx906

Is this true?

@Dmantri98 
> the data collection and model development for gfx1030 has not been done yet.

I can wait. BTW what is "data collection"? I thought that the Tuna database already contains enough data and it's actually about exporting this data, right?

Reply:
> I can wait. BTW what is "data collection"? I thought that the Tuna database already contains enough data and it's actually about exporting this data, right?

Actually, no. The data present in the MIOpen Dbs is very little to reliably train a Neural Network which can accurately predict a model. Therefore, we have another set of configs which augments the MIOpen data. Since it is larger and needs to be done for each GPU there is a considerable cost to collecting that data.

Therefore, as that data becomes available we can update MIOpen, however, we will not gate this PR on it.

Reply:
> > I don't believe we will be implementing the AI Heuristic for gfx906

> Is this true?

That is correct

Reply:
@JehandadKhan I see. Then let me provide instructions for performance testing on MI100 (on Monday) and analyze the resulting logs prior merging this.

Reply:
@JehandadKhan @Dmantri98 Sorry for delaying. I am preparing the scripts right now, ETA is tomorrow morning.

Reply:
@Dmantri98 @JehandadKhan Here is what I need: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1837#issuecomment-1312245164 Sorry for delaying ;(